src/server/game/Entities/Player/Player.cpp

```diff
@@ -18986,17 +18991,28 @@ bool Player::CheckInstanceLoginValid()
     {
         // cannot be in raid instance without a group
         if (!GetGroup())
-            return false;
+            return IsInstanceLoginGameMasterException();
```

Comment:
if player has gm command you will bypass all other checks, so i don't think this is not the right way to go

i guess you just need to adjust the gm level check which allows the player to enter in gm mode